### ISSUE TITLE
Pyyaml fails to install

### ISSUE THREAD

**Issue by withtwoemms** [scanapi/scanapi#675]
### Environment
- Operating System: macOS 11.7.10
- Python version: 3.12.0

### Description of the bug 
After installing `poetry` (1.7.1) with `pip install poetry` and then installing project dependencies with `poetry install`, the `pyyaml` dependency fails to install with output as follows:

```
$ poetry install
Installing dependencies from lock file

Package operations: 71 installs, 10 updates, 0 removals

  • Downgrading certifi (2023.11.17 -> 2023.5.7)
  • Downgrading charset-normalizer (3.3.2 -> 3.1.0)
  • Downgrading idna (3.6 -> 3.4)
  • Installing markupsafe (2.1.2)
  • Installing six (1.16.0)
  • Downgrading urllib3 (2.1.0 -> 2.0.2)
  • Installing alabaster (0.7.13)
  • Installing babel (2.12.1)
  • Installing docutils (0.17.1)
  • Installing imagesize (1.4.1)
  • Installing jinja2 (3.1.2)
  • Installing mdurl (0.1.2)
  • Downgrading packaging (23.2 -> 23.1)
  • Installing pygments (2.15.1)
  • Installing python-dateutil (2.8.2)
  • Downgrading requests (2.31.0 -> 2.30.0)
  • Installing setuptools (67.7.2)
  • Installing smmap (5.0.0)
  • Installing sniffio (1.3.0)
  • Installing snowballstemmer (2.2.0)
  • Installing sphinxcontrib-applehelp (1.0.2)
  • Installing sphinxcontrib-devhelp (1.0.2)
  • Installing sphinxcontrib-htmlhelp (2.0.0)
  • Installing sphinxcontrib-jsmath (1.0.1)
  • Installing sphinxcontrib-qthelp (1.0.3)
  • Installing sphinxcontrib-serializinghtml (1.1.5)
  • Installing anyio (3.6.2)
  • Installing arrow (1.2.1)
  • Installing click (8.0.3)
  • Downgrading distlib (0.3.8 -> 0.3.6)
  • Downgrading filelock (3.13.1 -> 3.12.0)
  • Installing gitdb (4.0.10)
  • Installing h11 (0.14.0)
  • Installing iniconfig (2.0.0)
  • Installing markdown-it-py (2.2.0)
  • Installing pbr (5.11.1)
  • Downgrading platformdirs (3.11.0 -> 3.5.0)
  • Installing pluggy (1.0.0)
  • Installing sh (1.14.2)
  • Installing sphinx (4.3.2)
  • Installing cfgv (3.3.1)
  • Installing coverage (7.2.5)
  • Installing freezegun (1.2.2)
  • Installing gitlint-core (0.18.0)
  • Installing gitpython (3.1.31)
  • Installing httpcore (0.17.0)
  • Installing identify (2.5.24)
  • Installing mccabe (0.7.0)
  • Installing mypy-extensions (1.0.0)
  • Installing nodeenv (1.8.0)
  • Installing pathspec (0.11.1)
  • Installing pycodestyle (2.9.1)
  • Installing pyflakes (2.5.0)
  • Installing pytest (7.3.1)
  • Installing pyyaml (6.0): Failed

  FileNotFoundError

  [Errno 2] No such file or directory: '/Users/withtwoemms/.pyenv/versions/scanapi/lib/python3.12/site-packages/virtualenv/create/via_global_ref/_virtualenv.py'

  at ~/.pyenv/versions/3.12.0/lib/python3.12/pathlib.py:1014 in open
      1010│         the built-in open() function does.
      1011│         """
      1012│         if "b" not in mode:
      1013│             encoding = io.text_encoding(encoding)
    → 1014│         return io.open(self, mode, buffering, encoding, errors, newline)
      1015│ 
      1016│     def read_bytes(self):
      1017│         """
      1018│         Open the file in bytes mode, read it, and close the file.

Cannot install pyyaml.

  • Installing responses (0.12.1)
  • Installing rich (13.3.5)
  • Installing sphinxcontrib-jquery (4.1)
  • Installing stevedore (3.5.2)
  • Installing types-urllib3 (1.26.25.12)
  • Installing typing-extensions (4.5.0)
  • Downgrading virtualenv (20.25.0 -> 20.23.0)
  ```

### Expected behavior ?
The expectation is that `poetry install` runs without issue installing all relevant dependencies.

### How to reproduce the the bug ?
1. ensure the use of python v3.12.0 (I used a virtualenv generated by `pyenv`--details available on request)
1. install poetry:
```
pip install poetry
```
1. install project dependencies:
```
poetry install
```

### Anthing else we need to know?
(not that I can think of at the moment...)

**Comment by withtwoemms** [comment]
shown to have been adequately addressed @ 6cb73602ec09d2b7e542f9b644b85c85b97a0ae8 giving this output after following the reproduction steps:
```
$ poetry install
Installing dependencies from lock file

Package operations: 71 installs, 10 updates, 0 removals

  • Downgrading certifi (2023.11.17 -> 2023.5.7)
  • Downgrading charset-normalizer (3.3.2 -> 3.1.0)
  • Downgrading idna (3.6 -> 3.4)
  • Installing markupsafe (2.1.2)
  • Installing six (1.16.0)
  • Downgrading urllib3 (2.1.0 -> 2.0.2)
  • Installing alabaster (0.7.13)
  • Installing babel (2.12.1)
  • Installing docutils (0.17.1)
  • Installing imagesize (1.4.1)
  • Installing jinja2 (3.1.2)
  • Installing mdurl (0.1.2)
  • Downgrading packaging (23.2 -> 23.1)
  • Installing pygments (2.15.1)
  • Installing python-dateutil (2.8.2)
  • Downgrading requests (2.31.0 -> 2.30.0)
  • Installing setuptools (67.7.2)
  • Installing smmap (5.0.0)
  • Installing sniffio (1.3.0)
  • Installing snowballstemmer (2.2.0)
  • Installing sphinxcontrib-applehelp (1.0.2)
  • Installing sphinxcontrib-devhelp (1.0.2)
  • Installing sphinxcontrib-htmlhelp (2.0.0)
  • Installing sphinxcontrib-jsmath (1.0.1)
  • Installing sphinxcontrib-qthelp (1.0.3)
  • Installing sphinxcontrib-serializinghtml (1.1.5)
  • Installing anyio (3.6.2)
  • Installing arrow (1.2.1)
  • Installing click (8.0.3)
  • Downgrading distlib (0.3.8 -> 0.3.6)
  • Downgrading filelock (3.13.1 -> 3.12.0)
  • Installing gitdb (4.0.10)
  • Installing h11 (0.14.0)
  • Installing iniconfig (2.0.0)
  • Installing markdown-it-py (2.2.0)
  • Installing pbr (5.11.1)
  • Downgrading platformdirs (3.11.0 -> 3.5.0)
  • Installing pluggy (1.0.0)
  • Installing sh (1.14.2)
  • Installing sphinx (4.3.2)
  • Installing cfgv (3.3.1)
  • Installing coverage (7.2.5)
  • Installing freezegun (1.2.2)
  • Installing gitlint-core (0.18.0)
  • Installing gitpython (3.1.31)
  • Installing httpcore (0.17.0)
  • Installing identify (2.5.24)
  • Installing mccabe (0.7.0)
  • Installing mypy-extensions (1.0.0)
  • Installing nodeenv (1.8.0)
  • Installing pathspec (0.11.1)
  • Installing pycodestyle (2.9.1)
  • Installing pyflakes (2.5.0)
  • Installing pytest (7.3.1)
  • Installing pyyaml (6.0.1)
  • Installing responses (0.12.1)
  • Installing rich (13.3.5)
  • Installing sphinxcontrib-jquery (4.1)
  • Installing stevedore (3.5.2)
  • Installing types-urllib3 (1.26.25.12)
  • Installing typing-extensions (4.5.0)
  • Downgrading virtualenv (20.25.0 -> 20.23.0)
  • Installing appdirs (1.4.4)
  • Installing bandit (1.7.5)
  • Installing curlify2 (1.0.1)
  • Installing flake8 (5.0.4)
  • Installing black (22.3.0)
  • Installing gitlint (0.18.0)
  • Installing httpx (0.24.0)
  • Installing isort (5.11.5)
  • Installing mypy (0.960)
  • Installing pre-commit (3.3.1)
  • Installing pytest-cov (4.0.0)
  • Installing pytest-freezegun (0.4.2)
  • Installing pytest-it (0.1.4)
  • Installing pytest-mock (3.2.0)
  • Installing requests-mock (1.10.0)
  • Installing sphinx-rtd-theme (1.2.0)
  • Installing types-pyyaml (6.0.12.9)
  • Installing types-requests (2.30.0.0)
  • Installing types-setuptools (68.0.0.3)

Installing the current project: scanapi (2.9.0)
```

will submit a PR shortly

**Comment by withtwoemms** [comment]
Upon investigation, [this issue on the poetry repo](https://github.com/python-poetry/poetry/issues/8287) confirms a known issue with `pyyaml` v6.0 being incompatible with [PEP 517](https://peps.python.org/pep-0517/). Remedy was to upgrade to `pyyaml` v6.0.1.
